src/bin/common/solving_loop.ml

```diff
@@ -351,10 +351,26 @@ let main () =
     State.create_key ~pipe:"" "produce_assignment"
   in
 
+  let steps_bound: int State.key =
+    State.create_key ~pipe:"" "steps-bound"
+  in
+
   let named_terms: DStd.Expr.term Util.MS.t State.key =
     State.create_key ~pipe:"" "named_terms"
   in
 
+  let set_steps_bound i st =
+    try
+      Steps.set_steps_bound i;
+      State.set steps_bound i st
+    with
+      Invalid_argument _ -> (* Raised by Steps.set_steps_bound*)
```

Comment:
I this is raised by `Steps.set_steps_bound` the `try .. with ..` block should only include `Steps.set_steps_bound` here.

Reply:
If set_steps_bound fails, we do not want `State.set` to be called. It was written initially when I put "recoverable_error" instead of "fatal_error" in the next line.

Reply:
In that case, the best way to write the code in modern OCaml is to use `match .. exception`:

```ocaml
match Steps.set_steps_bound i with
| () -> State.set steps_bound i st
| exception Invalid_argument _ ->
  fatal_error "..."
```

This helps controlling the scope of exception handling (I make the remark here because for standard exceptions such as `Not_found`, `Failure`, or `Invalid_argument` it is very easy to include an unrelated function in the `try .. with ..` block that could also raise that same exception and cause puzzling bugs).